go-selinux/selinux_linux.go

```diff
@@ -1095,3 +1104,115 @@ func dupSecOpt(src string) ([]string, error) {
 func disableSecOpt() []string {
 	return []string{"disable"}
 }
+
+func getDefaultContextFromReaders(c defaultSECtx) (string, error) {
+	if c.verifier == nil {
+		return "", fmt.Errorf("verifier function must not be nil")
+	}
+
+	context, err := newContext(c.scon)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to create label for %s", c.scon)
+	}
+
+	findUserInContext := func(r io.Reader) string {
+		fromRole := context["role"]
+		fromType := context["type"]
+		bufin := bufio.NewReader(r)
+
+		for done := false; !done; {
+			var (
+				line string
+				err  error
+			)
+
+			if line, err = bufin.ReadString('\n'); err != nil {
+				if err == io.EOF {
+					done = true
+				} else {
+					break
+				}
+			}
+			line = strings.TrimSpace(line)
+			if len(line) == 0 {
+				// Skip blank lines
+				continue
+			}
+			if line[0] == ';' || line[0] == '#' {
+				// Skip comments
+				continue
+			}
+
+			conns := strings.Fields(line)
+
+			lineConn := strings.TrimSpace(conns[0])
```

Comment:
Again, since you have used `strings.Fields` above, the `conns[0]` is guaranteed to not have extra spaces at the edges, so `strings.TrimSpace` here is obviously redundant.